### ISSUE TITLE
Traducción al inglés

### ISSUE THREAD

**Issue by fastfinge** [hxebolax/TranslateAdvanced#1]
Nota: Este texto fue traducido del inglés al español por Deep L. El texto en inglés se incluye al final. 

¡Hola! Desafortunadamente, no hablo español. Sin embargo, he leído el código fuente para traducir la interfaz. También he utilizado Deep L para ayudarme con la documentación. Yo estaba trabajando en la programación de un complemento similar, pero sólo llegué tan lejos como la adición de los servicios de Google y Deep L. Además, tu código es mucho mejor que el mío. Así que pensé que sería mejor traducir su addon en lugar de seguir trabajando en la mía. He estado usando esta traducción al inglés con éxito, así que creo que todo está correcto. Es posible que haya pequeños errores en la documentación, especialmente en las secciones como los agradecimientos, donde no pude comprobar el código fuente para asegurarme de que mi traducción es exacta a lo que escribiste. Adjuntaré el archivo zip a este tema. 
[en.zip](https://github.com/user-attachments/files/15780444/en.zip)

El texto en inglés es el siguiente.

Hello! Unfortunately, I do not speak Spanish. However, I have read the source code in order to translate the interface. I also used DeepL to help me with the documentation. I was working on programming a similar addon, but I only got as far as adding Google and DeepL services. Also, your code is much better than mine.  So I thought it would be better to translate your addon rather than continue to work on mine.  I have been using this English translation with success, so I think that everything is correct. It's possible there are small errors in the documentation, especially in the sections like acknowledgements, where I could not check the source code to make sure my translation is accurate to what you wrote. I will attach the zip file to this issue.  
[en.zip](https://github.com/user-attachments/files/15780444/en.zip)

**Comment by hxebolax** [comment]
Hello, thank you very much for your translation.
Let me tell you that I am implementing a language updater for the plugin in the next version.
This will make it easier for users to have the plugin updated to their language immediately without needing to download the entire plugin again.
My plan, which I have already used in other applications, is that when I receive a language update, I will upload it to the server. If the user checks for language updates, they will download it to the plugin.
This way, I avoid releasing a new version of the plugin every time I receive a language or language update.
Plugin updates will always include the latest translations received up to that point, and any new versions will be handled by the updater included in the plugin.
I plan to release a new version before the middle of next month with some new features, the language updater, and some fixes such as continuous reading.
For this reason, I ask for a little patience to see your translation reflected in the plugin.
I also ask that if you have no objection, you will be included in the translators' credits.
And now, with English added, if you commit to the update, in the next version I could present it to the international community.
Thank you again for your work.

**Comment by fastfinge** [comment]
Of course, I am happy to be in the translator's credits.  This is my first time doing translation work; is there somewhere I should sign up so that I can be notified when it's time to update the translation?  

I have many people in my extended family who speak a different language from me, so I use translation addons every day.  Thank you for your excellent work on this addon!  The features are excellent, and the design makes communicating with my family easier than ever.

**Comment by hxebolax** [comment]
Thank you for your words.
Actually, I believe that translations by translators should be based on the releases.
When I implement the updater, it will also be easy to get language updates.
Therefore, if you want to stay updated on when I release new versions, I would suggest subscribing to notifications here in the repository but setting them to custom and marking only releases.
That way, you will receive an email notification each time I publish an update of the plugin.
From there, the translator should see if anything has been updated in the documentation and the strings. I always include the .pot file with the plugin in the releases so that it can be used by translators and they can check for updates from that file.
I don't know what you work with, but I recommend Poedit, which can easily add new strings from the .pot file to the existing translation.
Best regards, and I hope to have a new version by the middle of next month, depending on how much time my work allows.

**Comment by fastfinge** [comment]
Thank you! Yes, I use poedit; I learned how from the NVDA documentation. I will set up notifications now.  Is it best to send you translation updates on github?

Also, would you like an English version of translates/languages.py?  I didn't translate that because it's a simple program, and I think that anyone using your addon can figure it out. Or they can just read the file and know what it does.

**Comment by hxebolax** [comment]
I prefer if you send me the translation here as you did.
It's convenient to obtain and I avoid accidentally deleting the email as has happened several times.
They send me a translation to my personal email and I unintentionally delete it in a hurry or without realizing it.
If you send it here, it will be on Github and it's easy for me to obtain the .zip and apply it.
 
Let me know if you need any further assistance!

**Comment by fastfinge** [comment]
Okay, I will do that! Thanks for the help, and I look forward to the updates!Sent from my iPhoneOn Jun 11, 2024, at 7:34 AM, hxebolax ***@***.***> wrote:﻿
I prefer if you send me the translation here as you did.
It's convenient to obtain and I avoid accidentally deleting the email as has happened several times.
They send me a translation to my personal email and I unintentionally delete it in a hurry or without realizing it.
If you send it here, it will be on Github and it's easy for me to obtain the .zip and apply it.
Let me know if you need any further assistance!

**Comment by fastfinge** [comment]
Hi! Congrats on the new release! Unfortunately, I can't find the translation string for the new menu item in any of the .poe files. Is there something I must do to make it show up for translation? I have pulled the latest from github and built the addon successfully with scons, but the .poe file did not change.

**Comment by hxebolax** [comment]
Hello, you need to run the following command in the root directory of the plugin:

scons pot

This will create a file named `TranslateAdvanced.pot` in the root directory of the plugin.
Now, you need to open this file with Poedit and go to the menu Translation / Properties.
In the dialog that opens, change the Source code charset combo box to UTF-8 (recommended).
Accept this dialog. Then close Poedit, and it will ask if you want to save the changes. Say yes.
Now, open your `nvda.po` file that you made for the English language with Poedit.
Once it's open, go to the menu Translation / Update from POT file… and in the dialog that opens, choose the file you generated with `scons pot` named `TranslateAdvanced.pot` located in the root of the code.
All untranslated strings will be added to your file.
However, I'm also providing you with the .pot file from the repository if you prefer to use it.

https://github.com/hxebolax/TranslateAdvanced/releases/download/2024.06.16/TranslateAdvanced.pot

If you have any questions, let me know.

**Comment by fastfinge** [comment]
Oh, got it! Thank you for your clear instructions, they were extremely helpful. Here is the updated translation.
[en.zip](https://github.com/user-attachments/files/15870238/en.zip)

**Comment by hxebolax** [comment]
Hello, I just arrived home and uploaded your translation.
Now, from the NVDA addon menu > Preferences > Advanced Translator, we can search for updates in the second item and the English language will be updated along with Ukrainian, which has also been added.
I appreciate your work and thank you very much.

**Comment by fastfinge** [comment]
Here's the English translation update:
[en.zip](https://github.com/user-attachments/files/15944640/en.zip)

**Comment by hxebolax** [comment]
Thank you very much for your work.
The update has already been uploaded, so we can search for updates from the plugin menu.
Thanks.

**Comment by fastfinge** [comment]
Here is the English translation update:
[en.zip](https://github.com/user-attachments/files/16949448/en.zip)

I'm sorry it took me so long; I've had a busy time at work recently.
